packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -118,37 +119,101 @@ export type EditableTreeOrPrimitive = EditableTree | PrimitiveValue;
 /**
  * EditableTree, but with these cases of unwrapping:
  * - primitives are unwrapped. See {@link EditableTreeOrPrimitive}.
- * - nodes with PrimaryField are unwrapped to just the primaryField. See `getPrimaryField`.
+ * - nodes with PrimaryField (see `getPrimaryField`) are unwrapped to {@link EditableField}.
  * - fields are unwrapped based on their schema's multiplicity. See {@link UnwrappedEditableField}.
  */
-export type UnwrappedEditableTree = EditableTreeOrPrimitive | readonly UnwrappedEditableTree[];
+export type UnwrappedEditableTree = EditableTreeOrPrimitive | EditableField;
 
 /**
- * A field of an {@link EditableTree}.
+ * A field of an {@link EditableTree} as an array-like sequence of unwrapped nodes (see {@link UnwrappedEditableTree}).
+ *
+ * The number of nodes depends on a field's multiplicity.
+ * When iterating, the nodes are read at once. Use index access to read the nodes "lazily".
+ * Use `getWithoutUnwrapping` to get the node without unwrapping.
  */
-export type EditableField = readonly [FieldSchema, FieldKey, readonly EditableTree[]];
+export interface EditableField extends ArrayLike<UnwrappedEditableTree> {
+    /**
+     * The `FieldSchema` of this field.
+     */
+    readonly fieldSchema: FieldSchema;
+
+    /**
+     * The `FieldKey` of this field.
+     */
+    readonly fieldKey: FieldKey;
+
+    /**
+     * Gets the type of the node by its index or the "field type" if index is not provided and this is the primary field.
+     * If the node is well-formed, it must follow its schema.
+     * @param index - if index is provided, returns the type of the node. Otherwise, if this field is the primary field,
+     * returns the type of the parent node this primary field belongs to.
+     * @param nameOnly - if true, returns only the type identifier.
+     */
+    getType(index?: number, nameOnly?: boolean): NamedTreeSchema | TreeSchemaIdentifier | undefined;
+
+    /**
+     * Gets the non-unwrapped node of this field by its index.
+     */
+    getWithoutUnwrapping(index: number): EditableTree;
```

Comment:
I would prefer to call it just `getNode`, as one always gets unwrapped node just by using an index access.